addons/l10n_eg_edi_eta/models/account_edi_format.py

```diff
@@ -0,0 +1,426 @@
+        request_payload = {}
+        headers = {'Content-Type': 'application/json',
+                   'Authorization': 'Bearer %s' % token}
+        try:
+            request_response = requests.request('GET', request_url, headers=headers, data=request_payload,
+                                                timeout=(5, 10))
+        except (ValueError, requests.exceptions.ConnectionError, requests.exceptions.MissingSchema, requests.exceptions.Timeout, requests.exceptions.HTTPError) as ex:
+            return {
+                'error': _('PDF Not Reached. \n%s' % ex)
+            }
+        _logger.warning('PDF Function Response %s.', request_response)
+        if request_response:
+            _logger.warning('PDF Function %s.', request_response.text)
+        if request_response.status_code in [404, 200]:
+            return request_response.content
+        else:
+            return {
+                'error': request_response.text
+            }
+
+    def _l10n_eg_invoice_discount(self, invoice):
+        return sum(line.discount for line in invoice.invoice_line_ids.filtered(lambda x: x.display_type not in ('line_section', 'line_note')))
+
+    def _l10n_eg_eta_prepare_eta_invoice(self, invoice):
+        def group_tax_retention(tax_values):
+            return {'l10n_eg_eta_code': tax_values['tax_id'].l10n_eg_eta_code.split('_')[0]}
+
+        date_string = invoice.invoice_date.strftime('%Y-%m-%dT%H:%M:%SZ')
+        edi_values = invoice._prepare_edi_vals_to_export()
+        grouped_taxes = invoice._prepare_edi_tax_details(
+            grouping_key_generator=group_tax_retention)['tax_details'].values()
+        eta_invoice = {
+            'issuer': self._l10n_eg_eta_prepare_address_data(invoice.journal_id.l10n_eg_branch_id, issuer=True,),
+            'receiver': self._l10n_eg_eta_prepare_address_data(invoice.partner_id, invoice_amount=invoice.amount_total),
+            'documentType': 'i' if invoice.move_type == 'out_invoice' else 'c' if invoice.move_type == 'out_refund' else 'd' if invoice.move_type == 'in_refund' else '',
+            'documentTypeVersion': '1.0',
+            'dateTimeIssued': date_string,
+            'taxpayerActivityCode': invoice.journal_id.l10n_eg_branch_id.l10n_eg_activity_type_id.code,
+            'internalID': invoice.name,
+        }
+        if invoice.move_type in ['out_refund', 'in_refund']:
+            eta_invoice.update({
+                'references': [invoice.reversed_entry_id.l10n_eg_uuid] if invoice.reversed_entry_id.l10n_eg_uuid else []
+            })
+        eta_invoice.update({
+            'invoiceLines': self._l10n_eg_eta_prepare_invoice_lines_data(invoice),
+            'totalDiscountAmount': invoice._get_amount_main_currency(edi_values['total_price_discount']),
+            'totalSalesAmount': invoice._get_amount_main_currency(edi_values['total_price_subtotal_before_discount']),
+            'netAmount': invoice._get_amount_main_currency(invoice.amount_untaxed),
```

Comment:
```suggestion
            'netAmount': invoice.amount_untaxed_signed,
```
Those total amounts on the invoices are what is stored in the accounting of Odoo and seems like only correct to send it that way to ETA as well.  At least if they are so smart to not complain about rounding errors.